pkg/rule/accessrule.go

```diff
@@ -136,9 +136,68 @@ func (a *Approval) IsRequired() bool {
 // I expect this will be different to what gets returned in the api response
 type Target struct {
 	// References the provider's unique ID
-	ProviderID   string            `json:"providerId"  dynamodbav:"providerId"`
-	ProviderType string            `json:"providerType"  dynamodbav:"providerType"`
-	With         map[string]string `json:"with"  dynamodbav:"with"`
+	ProviderID     string              `json:"providerId"  dynamodbav:"providerId"`
+	ProviderType   string              `json:"providerType"  dynamodbav:"providerType"`
+	With           map[string]string   `json:"with"  dynamodbav:"with"`
```

Comment:
Rather than having both `With` and `WithSelectable`, could we not simply have `With` as a `map[string][]string`? It looks like this could cause us issues in future if we need to remember to check `With` as well as `WithSelectable` (and we might default to using `With` and forget to also check `WithSelectable`).

My thinking here is that a rule with a singular target e.g. `{"account": "123}` could be represented as an array of length 1, e.g. `{"account": ["123"]}`.

Reply:
I note that this might require a DynamoDB migration to run - but that might be a good opportunity to figure out how to do that.

Reply:
I discussed it in my exploration doc, but my motives were compatibility, not wanting to have to work with migrations.
And I think it actually makes it a simpler distinction to be honest.

Though it probably looks a bit spaghetti code